icubam/config.py

```diff
@@ -47,13 +47,29 @@ def _preprocess(self, conf):
         result[k.lower()] = v
       else:
         result[k.lower()] = self._preprocess(v)
-
     return result
 
   def __getitem__(self, key: str):
     if key.upper() == key:
       return self.env.get(key)
-    return self.conf.get(key) if self.conf.has_key(key) else None
+    if self.conf.has_key(key):
+      return self.conf.get(key)
+    raise KeyError(f"key '{key}' does not exist!")
 
   def __getattr__(self, key: str):
-    return self[key]
+    raise_err = False
+    try:
+      res = self[key]
+    except KeyError:
+      raise_err = True
+    # Avoid raising an exception inside an exception for traceback
+    # readability.
+    if raise_err:
+      raise AttributeError(f"attribute '{key}' does not exist!")
```

Comment:
This is a bit awkward if we reuse `__getitem__` , but anyway the point is that we need to return  `AttributeError` for missing attibutes not None. Otherwise `hasattr(self.config, '__call__')` probably returns True (without it being a callable) and pickle is not able to process this object.